src/widget/weffectselector.cpp

```diff
@@ -36,6 +36,23 @@ void WEffectSelector::setup(const QDomNode& node, const SkinContext& context) {
                 << "EffectSelector node could not attach to effect slot.";
     }
 
+    // macOS displays a wide checkmark on the left of the drop down menu
+    // that displaces text and makes it get cut off on the right side.
+    // The checkmark icon can be hidden by styling WEffectSelector::indicator,
+    // but the space it occupies cannot be easily changed. So, set a negative
+    // left margin for WEffectSelector QAbstractItemView. However, that
+    // causes the text to get cut off on the left on KDE, so here is a horribly
+    // ugly hack to set the QSS margin from C++ in a platform dependent way.
+#ifdef _Q_OS_MAC
+        setStyleSheet(QString(
+            // Setting a more negative left margin than necessary puts the
+            // text up against the left side of the drop down menu but does
+            // not cut off the left side of the text on macOS.
```

Comment:
IIRC this was the case on my friend's Retina 2016 MacBook Pro, but an extreme negative left margin is cutting off the text on this 1440 x 900 2011 MacBook Air. At this point I am thinking we could add an element to specify the left margin on Mac so that the #ifdef in C++ can make it platform specific...
```
<EffectSelector>
...
  <MacLeftMargin>-24</MacLeftMargin>
</EffectSelector>
```

Reply:
I have another idea. We could make a platform-specific QSS file.

Reply:
We may run into more tricky platform dependent styling issues like this in the future, so I think platform-specific QSS files are a good idea. We could call them `style-mac.qss`, `style-windows.qss`, `style-linux.qss`.